### PR TITLE
IGNITE-16319: [ducktests] Get rid of duplicating test data generation application

### PR DESCRIPTION
**Pull request by skorotkov**
### Changes Description
There were two different applications for test data generation in the ignite-ducktests framework. Each used by the only test.

The change is to remove one of the applications (the **DataLoaderApplication** one) and modify affected tests accordingly to use the second (the **DataGenerationApplication** one).

---

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [X] There is a single JIRA ticket related to the pull request. 
- [X] The web-link to the pull request is attached to the JIRA ticket.
- [X] The JIRA ticket has the _Patch Available_ state.
- [X] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [X] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [X] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/ducktests/tests/ignitetest/tests/snapshot_test.py [hunk 1]
```diff
@@ -63,8 +63,14 @@ def snapshot_test(self, ignite_version):
         loader = IgniteApplicationService(
             self.test_context,
             loader_config,
-            java_class_name="org.apache.ignite.internal.ducktest.tests.snapshot_test.DataLoaderApplication",
-            params={"start": 0, "cacheName": self.CACHE_NAME, "interval": 500_000, "valueSizeKb": 1}
+            java_class_name="org.apache.ignite.internal.ducktest.tests.data_generation.DataGenerationApplication",
+            params={
```
Comment (nizhikov) [inline modules/ducktests/tests/ignitetest/tests/snapshot_test.py:67]: Let's have single params dictionary and modify it for second run

File: modules/ducktests/tests/ignitetest/tests/snapshot_test.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ class SnapshotTest(IgniteTest):
     """
     SNAPSHOT_NAME = "test_snapshot"
 
-    CACHE_NAME = "TEST_CACHE"
+    CACHE_NAME = "test-cache-1"
```
Comment (nizhikov) [inline modules/ducktests/tests/ignitetest/tests/snapshot_test.py:39]: Why this change?

File: modules/ducktests/tests/ignitetest/tests/snapshot_test.py [hunk 3]
```diff
@@ -63,8 +63,14 @@ def snapshot_test(self, ignite_version):
         loader = IgniteApplicationService(
```
Comment (nizhikov) [inline modules/ducktests/tests/ignitetest/tests/snapshot_test.py:63]: Can we invoke `preload_data` function here instead of manually running application

Comment (skorotkov) [reply modules/ducktests/tests/ignitetest/tests/snapshot_test.py]: Data loading code has been extracted to the separate class (DataLoader).  Both rebalance and snapshot tests use it now.

File: modules/ducktests/tests/ignitetest/tests/snapshot_test.py (hunk 1 above)
Comment (skorotkov) [reply modules/ducktests/tests/ignitetest/tests/snapshot_test.py]: Not applicable now after adding of the separate DataLoader class.
